### PR TITLE
chore: add markdown link check

### PR DESCRIPTION
**Pull request by legendecas**
Checks link liveness regularly.

### CODE REVIEW

**Comment by legendecas** [comment]
Well, seems like we do have a markdown linter that checks on markdown files but apparently it is not reporting on dead links :( https://github.com/open-telemetry/opentelemetry-js-api/blob/main/.github/workflows/lint.yaml#L37

File: .github/workflows/markdown-lint.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Link checking costs tens of minutes to run.
+# Runs everyday at 9 am.
+on:
+  # TODO: remove me before merging
+  pull_request:
+  schedule:
+    - cron: "0 9 * * *"
+
+jobs:
+  markdown-link-check:
```
Comment (vmarchaud) [inline .github/workflows/markdown-lint.yml:12]: we already have a job that link markdown here: https://github.com/open-telemetry/opentelemetry-js-api/blob/main/.github/workflows/lint.yaml#L17

Comment (legendecas) [reply .github/workflows/markdown-lint.yml:12]: Yeah, like what I'm commented above: https://github.com/open-telemetry/opentelemetry-js-api/pull/141#issuecomment-991925591 it doesn't check on files in docs/*.md.

Comment (legendecas) [reply .github/workflows/markdown-lint.yml:12]: I'd find another way to get it work instead of adding a new job.

Comment (vmarchaud) [reply .github/workflows/markdown-lint.yml:12]: > Yeah, like what I'm commented above: #141 (comment) it doesn't check on files in docs/*.md.

Sorry for some reason i didn't see the comment

**Review by dyladan — APPROVED** [review]

**Review by vmarchaud — APPROVED** [review]

**Pull request merged**